### DISCUSSION TITLE
Store Persistence: Is there a better way to do this?  Trying to "Partition" persist since store name can't be easily set in runtime.

### DISCUSSION THREAD

**Discussion by mrvfino** [pmndrs/zustand#2485] [Q&A] [answered]
For context, I'm developing a website builder and I'm trying to locally persist the specific editor state of a given site when the user saves.

**When Saving:** I iterate all the relevant keys, then create copies of them prefixed with an identifier.
**When Loading:** I just overwrite the current store values based , then reload the page on my component.

<img width="620" alt="Screenshot 2024-04-12 at 5 23 40 PM" src="https://github.com/pmndrs/zustand/assets/30117352/0442e850-2b72-48b0-8de3-83cd6b80d367">

**Comment by dbritto-dev** [comment, marked as answer]
@mrvfino have you ever tried to do the same with `partialize`?

**Reply by mrvfino** [reply]
Oh wait shit we have that? Thanks you, will read up on it now.

**Reply by mrvfino** [reply]
@dbritto-dev  Unfortunately this isn't the one i'm looking for. To clarify those keys are store names and not property names.

**Reply by dbritto-dev** [reply]
@mrvfino got it. Are you using the persist middleware or you are doing all the stuff manually?

**Reply by mrvfino** [reply]
@dbritto-dev I'm using the persist middleware. The manual part I do is overwriting the store with the saved editor state associated with the site the user wants to edit and refreshing to re-render stuff.
